src/mgmt/srv_drpc.c

```diff
@@ -786,10 +785,16 @@ void ds_mgmt_drpc_pool_set_prop(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 
 	switch (req->value_case) {
 	case MGMT__POOL_SET_PROP_REQ__VALUE_STRVAL:
-		D_ASPRINTF(out_str_val, "%s", req->strval);
-		if (out_str_val == NULL)
+		if (req->strval == NULL) {
+			D_ERROR("string value is NULL\n");
+			D_GOTO(out, rc = -DER_INVAL);
```

Comment:
For cases where the dRPC caller has mangled the struct, sometimes I use -DER_PROTO to indicate a violation of protocol expectations. This might be one of those cases.

Reply:
Updated, thanks.